src/assignment.js

```diff
@@ -158,38 +166,45 @@ function createAssignmentField(checkFor, item, fieldText, defaultState) {
   return assignmentsToggleField;
 }
 
-function getInitialState(item, defaultState) {
-  const itemState = localStorage.getItem(item);
-
+function getInitialState(item, defaultState, checkFor) {
+  const itemState = JSON.parse(localStorage.getItem(item))[1];
   if (itemState === "true") {
     return true;
   } else if (itemState === "false") {
     return false;
   }
 
-  localStorage.setItem(item, defaultState);
+  localStorage.setItem(item, JSON.stringify([checkFor,defaultState]));
   return defaultState;
 }
 
-function showHideAssignments(checkFor, item) {
+function showHideAssignments() {
   const assignmentsTable = document.getElementById("s_m_Content_Content_ExerciseGV");
   const assignmentsTBody = assignmentsTable.getElementsByTagName("tbody");
   const assignmentElements = assignmentsTBody[0].getElementsByTagName("tr");
 
-  let assignmentState = "table-row";
+  const items = JSON.parse(localStorage.getItem("items"))
```

Comment:
```suggestion
  const localStorageItems = JSON.parse(localStorage.getItem("items"));
  const items = localStorageItems.map(item => JSON.parse(localStorage.getItem(item)));
```
This must be used with the changes at lines ``+195 to +197``